### PR TITLE
Update docs theme version to 1.5

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** This PR updates the docs theme version to 1.5 as per https://sphinx-theme.scylladb.com/stable/upgrade/1-3-to-1-4.html and https://sphinx-theme.scylladb.com/stable/upgrade/1-4-to-1-5.html.
Underscores in source files are also replaced with hyphens due to a warning being thrown by Sphinx, causing the build to fail.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/scylladb/scylla-operator/issues/1201

### CODE REVIEW

**Comment by annastuchlik** [comment]
@rzetelskik The new version of the theme triggers a warning if an underscore is encountered in a file name. The following files should be renamed to use a hyphen instead of the underscore:
- known_issues
- nodeoperations/automatic_cleanup
- nodeoperations/maintenance_mode
- nodeoperations/replace_node
- nodeoperations/scylla_upgrade
- scylla_cluster_crd

**Comment by rzetelskik** [comment]
@annastuchlik yup, I've updated the PR just before your comment. Should be fine now.

File: docs/Makefile [hunk 1]
```diff
@@ -18,6 +18,7 @@ all: dirhtml
 
 clean:
 	rm -rf '$(BUILDDIR)'/*
+	rm -f poetry.lock
```
Comment (tnozicka) [inline docs/Makefile:21]: I don't think this belongs here. `clean` is meant to clean build files which can always be regenerated to the same output, `poetry.log` actually holds information (snapshot of all the moving references in a certain time), same as go.sum.

**Review by tnozicka — COMMENTED** [review]
modulo Makefile, this lgtm

File: docs/Makefile (hunk 1 above)
Comment (rzetelskik) [reply docs/Makefile:21]: I followed the [upgrade guide](https://sphinx-theme.scylladb.com/stable/upgrade/1-3-to-1-4.html), but personally I agree, it sort of defeats the purpose of even having the `.lock` file. @annastuchlik why is that considered a "breaking change" in the upgrade guide?

**Comment by annastuchlik** [comment]
Let me look into it with @dgarcia360 before I approve or suggest an update. Thanks.

File: docs/Makefile (hunk 1 above)
Comment (tnozicka) [reply docs/Makefile:21]: > I followed the upgrade guide
We don't follow their Makefile, nor CI for differences on opinions on what reproducibility means or make should look like. Totally fine to differ in `clean` target.

Having to remove `.lock` probably suggest something wrong is being done somewhere else but even so it should be one shot, not a part of regular cleanup.

Comment (rzetelskik) [reply docs/Makefile]: I agree. I removed the Makefile change.

**Comment by annastuchlik** [comment]
I'm unable to build the docs. It looks like more updates are required to the Makefile.

**Review by tnozicka — APPROVED** [review]
/lgtm

I'll give @annastuchlik a while to see if it works for her

**Comment by rzetelskik** [comment]
> I'm unable to build the docs. It looks like more updates are required to the Makefile. 

@annastuchlik can you elaborate? What did you try? What do you mean by being unable to build the docs?

**Comment by annastuchlik** [comment]
I get the following error:

```
Traceback (most recent call last):
  File "C:\Users\annstu\.poetry\bin\poetry", line 17, in <module>
    from poetry.console import main
ModuleNotFoundError: No module named 'poetry.console'
make: *** [Makefile:24: preview] Error 1
```

**Comment by rzetelskik** [comment]
> I get the following error:
> 
> ```
> Traceback (most recent call last):
>   File "C:\Users\annstu\.poetry\bin\poetry", line 17, in <module>
>     from poetry.console import main
> ModuleNotFoundError: No module named 'poetry.console'
> make: *** [Makefile:24: preview] Error 1
> ```

@annastuchlik please try the following:
```
podman run -it --rm -v="$( pwd ):/go/$( go list -m )" --workdir="/go/$( go list -m )/docs" -p 5500:5500 ubuntu:20.04 bash -c 'apt-get update && apt-get install -y curl python3 python3-distutils make git && ./hack/install-poetry.sh && $HOME/.poetry/bin/poetry install && make dirhtml SPHINXOPTS="--keep-going" && make redirects && make preview'
```
It's what our workflow does essentially.

**Comment by dgarcia360** [comment]
Hi @annastuchlik, @tnozicka, and @rzetelskik,

While these changes will build the docs as expected for prod, building the docs locally differs from how we do it in other projects.  I'm sharing some proposed modifications to the Makefile:

1. Modify the `POETRY `variable to ensure compatibility for Windows:

    ```
    POETRY        = poetry
    
    # For Windows users
    ifeq ($(OS),Windows_NT)
        POETRY = $(APPDATA)\Python\Scripts\poetry
    endif
    ```
    
2. Include the  `setup` step before executing commands to generate documentation. For example:

    ```
    # Preview commands
    .PHONY: preview
    preview: setup
	    $(POETRY) run sphinx-autobuild -b dirhtml $(ALLSPHINXOPTS) $(BUILDDIR)/dirhtml --port 5500
    ```
  
    This ensures that the latest version of the dependencies are installed before building the docs.
  
     If this suggestions don't align with your preferences, I'd like to propose an alternative. To provide more context, I'm sharing a README from another project that outlines their build process, which is different from the one provided by the Sphinx Theme:https://github.com/scylladb/cpp-driver/blob/dacab2ced2fa84d798723d13ecb8c8f658f28d30/README-dev.md
  
     You could create a similar "README-dev.md" within the docs directory, explaining the considerations needed to run the docs locally. For example: 
  
    ```md
    # Building the docs
    
    This project uses [Sphinx Theme](https://sphinx-theme.scylladb.com/) , but we have modified the base project to adapt it to our workflows.
    
    ## Local preview
    
    To build the documentation locally, run `make setup` first.
    
    Then, you can build and preview the documentation by following the steps outlined in the [Quickstart guide](https://sphinx-theme.scylladb.com/stable/getting-started/quickstart.html).
    
    ## Custom workflows 
    
    (optional) Explain other  customization. 
    ```

**Comment by rzetelskik** [comment]
@dgarcia360 thanks! 
Ad 2. I suppose we could add dirhtml/multiversion and redirects as prerequisites for preview and multiversionpreview.
I won't affect our workflow and if it'll align our Makefile with other projects, then I don't see why not. Same goes for Windows compatibility.

Having said that, it should go in a separate PR. @tnozicka any objections?

**Comment by rzetelskik** [comment]
@tnozicka @annastuchlik @dgarcia360 please see https://github.com/scylladb/scylla-operator/pull/1323

**Comment by tnozicka** [comment]
/lgtm

With readme shipping in #1323 are there any concerns about the content of this PR or can we land this bump?

**Comment by rzetelskik** [comment]
> With readme shipping in https://github.com/scylladb/scylla-operator/pull/1323 are there any concerns about the content of this PR or can we land this bump?

I don't think they interfere with each other so I believe they can land in any order.

**Comment by annastuchlik** [comment]
@rzetelskik @tnozicka Thanks!

**Pull request merged**